### PR TITLE
Finalize state after each call in callMany

### PR DESCRIPTION
**Pull request by ArielTM**
In `debug_traceCallMany` and `eth_callMany` the `IntraBlockState` is not being finalized after each replayed transaction and after each call. This can cause several problems:

- False gas consumption calculation. Specifically in `SSTORE` dynamic gas calculation being affected by "warm" and "cold" storage slots, and by "original" value comparison.
- Ability to call contracts that were `SELFDESTRUCT`ed in the replayed transactions or during previous calls, as the self-destruction has not been finalized.

see https://github.com/ledgerwatch/erigon/issues/6373 as an example.

This pull request adds a call to `FinalizeTx` after each `ApplyMessage` and `TraceTx`

### CODE REVIEW

File: cmd/rpcdaemon/commands/eth_callMany.go [hunk 1]
```diff
@@ -217,6 +217,9 @@ func (api *APIImpl) CallMany(ctx context.Context, bundles []Bundle, simulateCont
 		if err != nil {
 			return nil, err
 		}
+
+		_ = st.FinalizeTx(rules, state.NewNoopWriter())
```
Comment (AskAlexSharov) [inline cmd/rpcdaemon/commands/eth_callMany.go:221]: can you please check, if `SoftFinalise` also fixing your problem?

Comment (ArielTM) [reply cmd/rpcdaemon/commands/eth_callMany.go:221]: Seems like it doesn't

Comment (ArielTM) [reply cmd/rpcdaemon/commands/eth_callMany.go:221]: Let me know if any more changes are required :)

**Review by AskAlexSharov — APPROVED** [review]

**Comment by AskAlexSharov** [comment]
thank you. will be great if you can add some unit-test

**Pull request merged**